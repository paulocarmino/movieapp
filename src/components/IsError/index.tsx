import Button from "../Button"

export type IsErrorProps = {
  action?: () => void
}

const IsError = ({ action = () => { } }: IsErrorProps) => {
  return (
    <div className='my-10 text-center'>
      <svg xmlns="http://www.w3.org/2000/svg" className="mx-auto w-24 h-24 text-gray-100" fill="none" viewBox="0 0 24 24" stroke="currentColor" strokeWidth="2">
        <path strokeLinecap="round" strokeLinejoin="round" d="M12 8v4m0 4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z" />
      </svg>

      <h3 className="mt-2 text-sm font-medium text-gray-100">Error while searching!</h3>
      <p className="mt-1 text-sm text-gray-400">No results found for you search term.</p>

      <div className='flex justify-center items-center my-4 h-8'>
        <Button data-testid="action-button" onClick={() => action()}>Try again</Button>
      </div>
    </div>
  )
}
export default IsError
