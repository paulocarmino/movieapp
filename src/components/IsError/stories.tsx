import { Story, Meta } from '@storybook/react'
import IsError from '.'

export default {
  title: 'IsError',
  component: IsError,
  parameters: {
    backgrounds: {
      default: 'dark'
    }
  }
} as Meta

export const Default: Story = () => <IsError />
