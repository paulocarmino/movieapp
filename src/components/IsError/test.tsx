import { fireEvent, render, screen } from '@testing-library/react'

import IsError from '.'

describe('<IsError />', () => {
  it('should render the component', () => {
    const { container } = render(<IsError />)

    expect(screen.getByRole('heading', { name: /Error while searching!/i })).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should trigger action when click in button', () => {
    const actionMock = jest.fn();
    const { getByTestId } = render(<IsError action={actionMock} />)

    fireEvent.click(getByTestId('action-button'))
    expect(actionMock).toBeCalled()
  })
})
