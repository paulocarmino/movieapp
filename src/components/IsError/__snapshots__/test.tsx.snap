// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IsError /> should render the component 1`] = `
<div
  class="my-10 text-center"
>
  <svg
    class="mx-auto w-24 h-24 text-gray-100"
    fill="none"
    stroke="currentColor"
    stroke-width="2"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12 8v4m0 4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
  <h3
    class="mt-2 text-sm font-medium text-gray-100"
  >
    Error while searching!
  </h3>
  <p
    class="mt-1 text-sm text-gray-400"
  >
    No results found for you search term.
  </p>
  <div
    class="flex justify-center items-center my-4 h-8"
  >
    <button
      class="inline-flex items-center py-2 px-3 text-sm leading-4 text-gray-100 hover:bg-gray-800 rounded-md border border-gray-100 focus:outline-none shadow-sm"
      data-testid="action-button"
      type="button"
    >
      Try again
    </button>
  </div>
</div>
`;
